gtsam/geometry/Unit3.cpp

```diff
@@ -236,7 +226,7 @@ Vector2 Unit3::errorVector(const Unit3& q, OptionalJacobian<2, 2> H_p,
 /* ************************************************************************* */
 double Unit3::distance(const Unit3& q, OptionalJacobian<1, 2> H) const {
   Matrix2 H_xi_q;
-  const Vector2 xi = error(q, H ? &H_xi_q : nullptr);
+  const Vector2 xi = errorVector(q, boost::none, H ? &H_xi_q : nullptr);
```

Comment:
I think this PR should not modify Unit3, right?

Reply:
I've reverted the changes to `Unit3::error()` that I made